packages/p2p/src/components/composite-calendar/composite-calendar.jsx

```diff
@@ -0,0 +1,201 @@
+    loading: () => null,
+    render(loaded, props) {
+        const Component = loaded.default;
+        return <Component {...props} />;
+    },
+});
+
+const CompositeCalendar = props => {
+    const { onChange, to, from } = props;
+    const {
+        ui: { current_focus, setCurrentFocus },
+    } = useStore();
+
+    const [show_to, setShowTo] = React.useState(false);
+    const [show_from, setShowFrom] = React.useState(false);
+    const list = [
+        {
+            value: 'all_time',
+            label: localize('All time'),
+            onClick: () => selectDateRange(),
+            duration: 0,
+        },
+        {
+            value: 'today',
+            label: localize('Today'),
+            onClick: () => selectDateRange(1),
+            duration: 1,
+        },
+        {
+            value: 'last_7_days',
+            label: localize('Last 7 days'),
+            onClick: () => selectDateRange(7),
+            duration: 7,
+        },
+        {
+            value: 'last_30_days',
+            label: localize('Last 30 days'),
+            onClick: () => selectDateRange(30),
+            duration: 30,
+        },
+        {
+            value: 'last_60_days',
+            label: localize('Last 60 days'),
+            onClick: () => selectDateRange(60),
+            duration: 60,
+        },
+        {
+            value: 'last_quarter',
+            label: localize('Last quarter'),
+            onClick: () => selectDateRange(90),
+            duration: 90,
+        },
+    ];
+
+    const wrapper_ref = React.useRef();
+
+    useOnClickOutside(wrapper_ref, event => {
+        event.stopPropagation();
+        event.preventDefault();
+        hideCalendar();
+    });
+
+    const selectDateRange = new_from => {
+        hideCalendar();
+        applyBatch({
+            from: new_from ? toMoment().startOf('day').subtract(new_from, 'day').add(1, 's').unix() : null,
+            to: toMoment().endOf('day').unix(),
+            is_batch: true,
+        });
+    };
+
+    const getToDateLabel = () => {
+        const date = epochToMoment(to);
+        return daysFromTodayTo(date) === 0 ? localize('Today') : date.format('MMM, DD YYYY');
+    };
+
+    const getFromDateLabel = () => {
+        const date = epochToMoment(from);
+        return from ? date.format('MMM, DD YYYY') : '';
+    };
+
+    const hideCalendar = () => {
+        setShowFrom(false);
+        setShowTo(false);
+    };
+
+    const showCalendar = e => {
+        if (e === 'from') {
+            setShowFrom(true);
+        }
+        if (e === 'to') {
+            setShowTo(true);
+        }
+    };
+
+    const setToDate = date => {
+        updateState('to', epochToMoment(date).endOf('day').unix());
+    };
+
+    const setFromDate = date => {
+        updateState('from', date);
+        hideCalendar();
+    };
+
+    const updateState = (key, value) => {
+        apply(key, value);
+        hideCalendar();
+    };
+
+    const applyBatch = values => {
```

Comment:
do we really need this applyBatch function since it's just onChange? an example would be in the apply function:

```
onChange({[key]: value});
```

Reply:
same in the selectDateRange function